test/utils/format/formatSmart.spec.ts

```diff
@@ -0,0 +1,184 @@
+  })
+
+  test('12,345 units', async () => {
+    const amount = new BN('12345000000')
+    expect(formatSmart(amount, 6)).toEqual('12,345')
+  })
+
+  test('4,567,890.123444 units, use precision 2', async () => {
+    const amount = new BN('4567890123444')
+    expect(formatSmart(amount, 6)).toEqual('4,567,890.12')
+  })
+})
+
+describe('Tokens with precision 0', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1')
+    expect(formatSmart(amount, 0)).toEqual('1')
+  })
+
+  test('12,345 units', async () => {
+    const amount = new BN('12345')
+    expect(formatSmart(amount, 0)).toEqual('12,345')
+  })
+
+  test('4,567,890 units', async () => {
+    const amount = new BN('4567890')
+    expect(formatSmart(amount, 0)).toEqual('4,567,890')
+  })
+})
+
+describe('Tokens with precision 2', () => {
+  test('1 unit', async () => {
+    const amount = new BN('100')
+    expect(formatSmart(amount, 2)).toEqual('1')
+  })
+
+  test('12.34 units', async () => {
+    const amount = new BN('1234')
+    expect(formatSmart(amount, 2)).toEqual('12.34')
+  })
+
+  test('1,234,567.89 units', async () => {
+    const amount = new BN('123456789')
+    expect(formatSmart(amount, 2)).toEqual('1,234,567.89')
+  })
+
+  test('1,234,567.8 units', async () => {
+    const amount = new BN('123456780')
+    expect(formatSmart(amount, 2)).toEqual('1,234,567.8')
+  })
+})
+
+describe('2 decimals', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1000000')
+    expect(formatSmart({ amount, precision: 6, decimals: 2 })).toEqual('1')
+  })
+
+  test('12.34 units', async () => {
+    const amount = new BN('12340000')
+    expect(formatSmart({ amount, precision: 6, decimals: 2 })).toEqual('12.34')
+  })
+
+  test('4,567,890.123333 units, round down', async () => {
+    const amount = new BN('4567890123333')
+    expect(formatSmart({ amount, precision: 6, decimals: 2 })).toEqual('4,567,890.12')
+  })
+
+  test('4,567,890.125555 units, round up', async () => {
+    const amount = new BN('4567890125555')
+    expect(formatSmart({ amount, precision: 6, decimals: 2 })).toEqual('4,567,890.13')
+  })
+})
+
+describe('0 decimals', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1000000')
+    expect(formatSmart({ amount, precision: 6, decimals: 0 })).toEqual('1')
+  })
+
+  test('12.34 units', async () => {
+    const amount = new BN('12340000')
+    expect(formatSmart({ amount, precision: 6, decimals: 0 })).toEqual('12')
+  })
+
+  test('4,567,890.123333 units, round down', async () => {
+    const amount = new BN('4567890123333')
+    expect(formatSmart({ amount, precision: 6, decimals: 0 })).toEqual('4,567,890')
+  })
+
+  test('4,567,890.125555 units, round up', async () => {
+    const amount = new BN('4567890125555')
+    expect(formatSmart({ amount, precision: 6, decimals: 0 })).toEqual('4,567,890')
+  })
+})
+
+describe('Big amounts', () => {
+  // TODO: Considering showing K,M,B,...
```

Comment:
Can you make this comment a issue if it really make sense?

Reply:
this should have been removed